pkg/controller/direct/sql/sqlinstance_controller.go

```diff
@@ -93,14 +97,26 @@ func (m *sqlInstanceModel) AdapterForObject(ctx context.Context, kube client.Rea
 		return nil, err
 	}
 
-	return &sqlInstanceAdapter{
+	adapter := &sqlInstanceAdapter{
 		projectID:           projectID,
 		resourceID:          resourceID,
 		desired:             obj.DeepCopy(),
 		sqlOperationsClient: gcpClient.sqlOperationsClient(),
 		sqlInstancesClient:  gcpClient.sqlInstancesClient(),
 		sqlUsersClient:      gcpClient.sqlUsersClient(),
-	}, nil
+	}
+
+	unmanaged, ok := obj.GetAnnotations()["cnrm.cloud.google.com/unmanaged"]
+	if ok && unmanaged != "" {
+		unmanagedFields := strings.Split(unmanaged, ",")
+		for _, field := range unmanagedFields {
+			if field != unmanageEditionAnnotationValue {
+				return nil, fmt.Errorf("unmanaging field `%s` is not supported, supported fields are: %s", field, unmanageEditionAnnotationValue)
+			}
```

Comment:
Would be nice to dedupe. Otherwise we should add a test with a duplicate to ensure it doesn't cause problems.